### ISSUE TITLE
Fitting crash due to effect exceptions

### ISSUE THREAD

**Issue by blitzmann** [pyfa-org/Pyfa#894]
We need to look into a solution for when there is an exception caused by an effect - currently, it kills the entire fitting window, which is great as it definitely tells us that something happened instead of waiting to find out there was an error. But preferably, it would continue on.

Possibly a try...catch in the top-most Effect class. We can even change the "Active" flag to "Error" so that the user notices this in the items stats > effects.

I also need to look at #247 again and make sure this works. I would want it to pop up with the error when the effect runs into an exception, but I do not want it to stop execution. Could be tricky.

**Comment by Ebag333** [comment]
>We can even change the "Active" flag to "Error" so that the user notices this in the items stats > effects.

I like this idea. I don't think that users will notice it, but it'd be super useful for devs. 

Try/except when we run effects seems reasonable. That should be pretty easy to implement.

**Comment by blitzmann** [comment]
Yeah, it should be pretty easy to catch exceptions when the handler is defined in `eos/gamedata.py`. For example replace the current handler with our own function that takes `*args, **kwargs` and call the real handler with those same args, and wrap that in a try...catch.

The problem would come from the GUI reporting since we can't spawn that in eos. I don't remember how I gt it to work - it may very well simply listen for output on stderr, in which case this would be pretty easy all around. I'll take a look at it later.
